qt/widgets/common/test/MuonPeriodInfoTest.h

```diff
@@ -0,0 +1,107 @@
+  static void destroySuite(MuonPeriodInfoTest *suite) { delete suite; }
+  void setUp() override { m_periodInfo = std::make_unique<MuonPeriodInfo>(); }
+  void tearDown() override { m_periodInfo.reset(); }
+
+  void assertRowValues(const int row, const std::string &name, const std::string &type, const std::string &DAQNumber,
+                       const std::string &frames, const std::string &totalFrames, const std::string &counts,
+                       const std::string &tag) {
+    const auto table = m_periodInfo->getTable();
+    TS_ASSERT_EQUALS(std::to_string(row + 1), table->item(row, 0)->text().toStdString()); // Period Count
+    TS_ASSERT_EQUALS(name, table->item(row, 1)->text().toStdString());
+    TS_ASSERT_EQUALS(type, table->item(row, 2)->text().toStdString());
+    TS_ASSERT_EQUALS(DAQNumber, table->item(row, 3)->text().toStdString());
+    TS_ASSERT_EQUALS(frames, table->item(row, 4)->text().toStdString());
+    TS_ASSERT_EQUALS(totalFrames, table->item(row, 5)->text().toStdString());
+    TS_ASSERT_EQUALS(counts, table->item(row, 6)->text().toStdString());
+    TS_ASSERT_EQUALS(tag, table->item(row, 7)->text().toStdString());
+  }
+
+  void test_that_the_table_is_empty_on_initialization() { TS_ASSERT_EQUALS(true, m_periodInfo->isEmpty()); }
+
+  void test_clear() {
+    auto table = m_periodInfo->getTable();
+    table->insertRow(0);
+    table->insertRow(1);
+    TS_ASSERT_EQUALS(false, m_periodInfo->isEmpty());
+    m_periodInfo->clear();
+    TS_ASSERT_EQUALS(true, m_periodInfo->isEmpty());
+    TS_ASSERT_EQUALS(0, m_periodInfo->getDAQCount());
+    TS_ASSERT_EQUALS(-1, m_periodInfo->getNumberOfSequences());
+    TS_ASSERT_EQUALS("Period Information for Run(s) ", m_periodInfo->getWidgetTitleRuns());
+  }
+
+  void test_addPeriodToTable() {
+    m_periodInfo->addPeriodToTable("state 1 dwell", "2", "10", "200", "25", "1");
+    m_periodInfo->addPeriodToTable("state 1", "1", "50", "1000", "25", "2");
+    assertRowValues(0, "state 1 dwell", "DWELL", "-", "10", "200", "-", "0001");
+    assertRowValues(1, "state 1", "DAQ", "1", "50", "1000", "25", "0010");
+    TS_ASSERT_EQUALS(2, m_periodInfo->getTable()->rowCount());
+    TS_ASSERT_EQUALS(false, m_periodInfo->isEmpty());
+  }
+
+  void test_addPeriodToTable_with_bad_tag() {
+    m_periodInfo->addPeriodToTable("state 1 dwell", "2", "10", "200", "25", "tag");
+    assertRowValues(0, "state 1 dwell", "DWELL", "-", "10", "200", "-", "Not found");
+  }
+
+  void test_DAQ_count_increases_as_expected() {
+    m_periodInfo->addPeriodToTable("state 1", "1", "10", "200", "25", "1");
+    TS_ASSERT_EQUALS(1, m_periodInfo->getDAQCount());
+    m_periodInfo->addPeriodToTable("state 1 dwell", "2", "10", "200", "25", "1");
+    TS_ASSERT_EQUALS(1, m_periodInfo->getDAQCount());
+    m_periodInfo->addPeriodToTable("state 1", "1", "10", "200", "25", "1");
+    m_periodInfo->addPeriodToTable("state 1", "1", "10", "200", "25", "1");
+    TS_ASSERT_EQUALS(3, m_periodInfo->getDAQCount());
```

Comment:
what trigged the DAQ count to increase?

Reply:
Adding a DAQ period to the table. the second argument is the type of period (1=DAQ, 2=DWELL). Here we added two daq periods as they both have the second argument as 1

Reply:
Could we have comment explaining that?